wrapper.ts

```diff
@@ -1,360 +1,159 @@
-    }
+function translateSources (input) {
+  const sources = {};
 
-    if (compileJSONMulti !== null) {
-      return translateOutput(compileJSONMulti(JSON.stringify({ sources: sources }), isOptimizerEnabled(input)), libraries);
+  for (const source in input.sources) {
+    if (input.sources[source].content !== null) {
+      sources[source] = input.sources[source].content;
+    } else {
+      // force failure
+      return null;
     }
+  }
 
-    // Try our luck with an ancient compiler
-    if (compileJSON !== null) {
-      if (Object.keys(sources).length !== 1) {
-        return formatFatalError('Multiple sources provided, but compiler only supports single input.');
-      }
-      return translateOutput(compileJSON(sources[Object.keys(sources)[0]], isOptimizerEnabled(input)), libraries);
-    }
+  return sources;
+}
 
-    return formatFatalError('Compiler does not support any known interface.');
-  };
+function librariesSupplied (input) {
+  if (!isNil(input.settings)) return input.settings.libraries;
+}
 
-  return {
-    version: version,
-    semver: versionToSemver,
-    license: license,
-    lowlevel: {
-      compileSingle: compileJSON,
-      compileMulti: compileJSONMulti,
-      compileCallback: compileJSONCallback,
-      compileStandard: compileStandard
-    },
-    features: {
-      legacySingleInput: compileJSON !== null,
-      multipleInputs: compileJSONMulti !== null || compileStandard !== null,
-      importCallback: compileJSONCallback !== null || compileStandard !== null,
-      nativeStandardJSON: compileStandard !== null
-    },
-    compile: compileStandardWrapper,
-    // Loads the compiler of the given version from the github repository
-    // instead of from the local filesystem.
-    loadRemoteVersion: function (versionString, cb) {
-      const mem = new MemoryStream(null, { readable: false });
-      const url = 'https://binaries.soliditylang.org/bin/soljson-' + versionString + '.js';
-      https.get(url, function (response) {
-        if (response.statusCode !== 200) {
-          cb(new Error('Error retrieving binary: ' + response.statusMessage));
-        } else {
-          response.pipe(mem);
-          response.on('end', function () {
-            // Based on the require-from-string package.
-            const soljson = new Module();
-            soljson._compile(mem.toString(), 'soljson-' + versionString + '.js');
-            if (module.parent && module.parent.children) {
-              // Make sure the module is plugged into the hierarchy correctly to have parent
-              // properly garbage collected.
-              module.parent.children.splice(module.parent.children.indexOf(soljson), 1);
-            }
+function translateOutput (outputRaw, libraries) {
+  try {
+    const parsedOutput = JSON.parse(outputRaw);
+    const output = translate.translateJsonCompilerOutput(parsedOutput, libraries);
```

Comment:
This `try` block is now too broad. It's only supposed to cover `JSON.parse()`. An exception on one of the other lines does not necessarily mean that invalid JSON was returned.

Reply:
sure I can simplify it.